integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+
+		chainSelToRPCURL[details.ChainSelector] = chain.HTTPRPCs[0].Internal
+	}
+
+	reorgDestClient := ctf_client.NewRPCClient(chainSelToRPCURL[destSelector], nil)
+
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, sourceSelector, destSelector, false)
+
+	// wait for log poller filters to get registered.
+	l.Info().Msg("waiting for log poller filters to get registered")
+	time.Sleep(15 * time.Second)
+	reorgingMsgEvent := testhelpers.TestSendRequest(t, e.Env, state, sourceSelector, destSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.
+		Info().
+		Str("messageID", hexutil.Encode(reorgingMsgEvent.Message.Header.MessageId[:])).
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Str("messageBlockHash", reorgingMsgEvent.Raw.BlockHash.String()).
+		Msgf("sent CCIP message that whose commit report will re-org on dest before getting finalized")
+
+	// expect the commit to still go through for the message.
+	reportEvent, err := testhelpers.ConfirmCommitWithExpectedSeqNumRange(
+		t,
+		sourceSelector,
+		e.Env.Chains[destSelector],
+		state.Chains[destSelector].OffRamp,
+		nil, // startBlock
+		ccipocr3.NewSeqNumRange(1, 1),
+		false, // enforceSingleCommit
+	)
+	require.NoError(t, err)
+
+	l.
+		Info().
+		Uint64("reportBlockNumber", reportEvent.Raw.BlockNumber).
+		Str("reportBlockHash", reportEvent.Raw.BlockHash.String()).
+		Msg("got commit report on dest, preparing to re-org it")
+
+	// re-org the dest chain less than finality blocks.
+	const reorgDepth = 7
+	var minChainBlockNumberBeforeReorg = reportEvent.Raw.BlockNumber + reorgDepth - 1
+	// Wait for chain to progress
+	require.Eventually(t, func() bool {
+		bn, err := reorgDestClient.BlockNumber()
+		require.NoError(t, err)
+		l.Info().
+			Int64("blockNumber", bn).
+			Uint64("targetBlockNumber", minChainBlockNumberBeforeReorg).
+			Msg("Waiting for chain to progress above target block number")
+		return bn >= int64(minChainBlockNumberBeforeReorg) //nolint:gosec // this will not overflow in the current test
+	}, 1*time.Minute, 500*time.Millisecond, "timeout exceeded: chain did not progress above the target block number")
+
+	// Run reorg below finality depth
+	l.Info().
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Int("reorgDepth", reorgDepth).
+		Uint64("sourceChainSelector", sourceSelector).
+		Msg("starting blockchain reorg on Simulated Geth chain")
+	err = reorgDestClient.GethSetHead(reorgDepth)
+	require.NoError(t, err, "error starting blockchain reorg on Simulated Geth chain")
+
+	bnAfterReorg, err := reorgDestClient.BlockNumber()
+	require.NoError(t, err, "error getting block number after reorg")
+
+	l.Info().Int64("blockNumberAfterReorg", bnAfterReorg).Msg("block number after reorg")
```

Comment:
can we put this whole thing into a wrapper func?

Reply:
Yeah gonna refactor this